presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedStatisticsCacheManager.java

```diff
@@ -74,7 +74,7 @@ public Map<PlanNodeWithHash, HistoricalPlanStatistics> loadAll(Iterable<? extend
                             return emptyResult;
                         }
                         Map<PlanNodeWithHash, HistoricalPlanStatistics> statistics = new HashMap<>(historyBasedPlanStatisticsProvider.get().getStats(ImmutableList.copyOf(keys), timeoutInMilliSeconds));
-                        if (statistics.isEmpty()) {
+                        if (statistics.isEmpty() || statistics.entrySet().stream().allMatch(x -> x.getValue().equals(empty()))) {
```

Comment:
Empty map and all values are empty statistics are equivalently the same, i.e. no history